python/hsfs/core/feature_group_engine.py

```diff
@@ -83,6 +88,11 @@ def insert(
         storage,
         write_options,
     ):
+        validation_id = None
+        if feature_group.validation_type != "NONE":
```

Comment:
if the `validation_type` is only coming from the backend, this is ok. but if the user is able to set it, please sanitize it with `upper()`

Reply:
I was thinking of making validation_type be an enum, like the rule_name.py. Then we can avoid the upper I guess

Reply:
So the client will get a key error when we use an Enum so I think that's good.
```
from enum import Enum, auto

class ValidationType(Enum):

    STRICT = auto()
    WARNING = auto()
    ALL = auto()
    NONE = auto()

bla = ValidationType["STRICt"]
An error was encountered:
'STRICt'
Traceback (most recent call last):
  File "/srv/hops/anaconda/envs/theenv/lib/python3.7/enum.py", line 357, in __getitem__
    return cls._member_map_[name]
KeyError: 'STRICt'
```

Reply:
:+1: I've never used Enums in python, but looks nice